File: eclipse/02-Exercicio/src/br/com/fiap/bean/ContaCorrente.java
package br.com.fiap.bean;

import java.util.Calendar;

import br.com.fiap.exception.SaldoInsuficienteException;

public class ContaCorrente extends Conta {
	private TipoConta tipo;

	public ContaCorrente(int agencia, int numero, Calendar dataAbertura, double saldo, TipoConta tipo) {
		super(agencia, numero, dataAbertura, saldo);
		this.tipo = tipo;
	}

	@Override
	public void depositar(double valor) {
		saldo += valor;
	}

	@Override
	public void retirar(double valor) throws SaldoInsuficienteException {
		if(tipo == TipoConta.COMUN && saldo < valor) {
			throw new SaldoInsuficienteException("Conta comum não pode ter saldo negativo.");
		}
		saldo -= valor;
	}

	public TipoConta getTipo() {
		return tipo;
	}

	public void setTipo(TipoConta tipo) {
		this.tipo = tipo;
	}
}

File: .NET/aula2/Fiap.Bank/Models/ContaCorrente.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Fiap.Bank.Models
{
    internal class ContaCorrente : Conta
    {
        TipoConta Tipo;

        public ContaCorrente(int agencia, int numero, DateTime dataAbertura, decimal saldo) : base(agencia, numero, dataAbertura, saldo)
        {
            Tipo = TipoConta.Comum;
        }

        public override void Depositar(decimal valor)
        {
            this.Saldo += valor;
            Console.WriteLine("Saldo depositado com sucesso. Saldo atual: R$", this.Saldo);
        }

        public override void Retirar(decimal valor)
        {
            if(this.Saldo >= valor && this.Tipo == TipoConta.Comum)
            {
                this.Saldo -= valor;
                Console.WriteLine("Saldo retirado no valor: R$", valor);
            } else
            {
                Console.WriteLine("Saldo insuficiente.");
            }
        }
    }
}
